### PR TITLE
Address a few MISRA 2.2 violations in FreeRTOS_IP.c

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
These changes bring in the changes required for addressing some of the MISRA rule 2.2 violations in FreeRTOS_IP.c.
The rule dictates that there shall be no dead code. When `ipconfigPRINTF` is not defined, then the debugging related variables seem to be unused. Use of ( void ) explicitly makes use of these variables while keeping the logic of code unchanged.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c [hunk 1]
```diff
@@ -2105,6 +2105,7 @@ uint8_t ucProtocol;
 		if( xResult != pdPASS )
 		{
 			FreeRTOS_printf( ( "xCheckSizeFields: location %ld\n", xLocation ) );
+			( void ) xLocation;
```
Comment (gkwicker) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c:2108]: I would add a comment here as to why this statement exists; same for line 2383.

Comment (AniruddhaKanhere) [reply FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c:2112]: Done, thanks!

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c [hunk 2]
```diff
@@ -2379,6 +2384,11 @@ BaseType_t location = 0;
 		( usChecksum == ipINVALID_LENGTH ) )
 	{
 		FreeRTOS_printf( ( "CRC error: %04x location %ld\n", usChecksum, location ) );
+		
+		/* If FreeRTOS_printf is not defined, not using xLocation will be a violation of MISRA
```
Comment (yanjos-dev) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c:2388]: Can you change "xLocation" to be "location" to match the variable name?

Comment (AniruddhaKanhere) [reply FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c]: I was supposed to fix that in the [previous commit](https://github.com/FreeRTOS/FreeRTOS/pull/230/commits/a3e02c19656da2732445356bc5bd45a0601f5b7b). Thanks for spotting that mistake.
It is fixed now.

**Pull request merged**